external/urlfrontier/src/main/java/org/apache/stormcrawler/urlfrontier/Spout.java

```diff
@@ -131,12 +144,19 @@ protected void populateBuffer() {
                 maxBucketNum,
                 maxURLsPerBucket);
 
-        GetParams request =
+        GetParams.Builder builder =
                 GetParams.newBuilder()
                         .setMaxUrlsPerQueue(maxURLsPerBucket)
-                        .setMaxQueues(maxBucketNum)
                         .setDelayRequestable(delayRequestable)
-                        .build();
+                        .setMaxQueues(maxBucketNum);
+
+        if (URLFRONTIER_ANY_CRAWL_ID.equals(globalCrawlID)) {
+            builder.setAnyCrawlID(anyCrawlID);
```

Comment:
we can set it explicitly but given it is what it does by default we can probably get away with just adding a comment.
We could then remove the constant  `URLFRONTIER_ANY_CRAWL_ID` as well as `anyCrawlID` and simplify the logic?
Said differently, if a value is set use it, otherwise rely on the default behaviour?

Reply:
Hi @jnioche , ok I removed references to anyCrawlId and populate the crawlId only if found in the configuration